openedx/features/cms/tests/test_helpers.py

```diff
@@ -0,0 +1,759 @@
+            org='org',
+            number='number',
+            run='test_dest',
+            display_name='destination course',
+            start=datetime(2019, 10, 1, tzinfo=UTC)
+        )
+
+        mock_calculate_date_by_delta.side_effect = [datetime(2020, 10, 1, tzinfo=UTC),
+                                                    datetime(2019, 10, 1, tzinfo=UTC),
+                                                    datetime(2019, 10, 10, tzinfo=UTC)]
+
+        helpers.set_rerun_schedule_dates(re_run_course, mock.Mock(), self.user)
+
+        # Testing course end, enrollment start and enrollment end date
+        updated_dest_course = get_course_by_id(re_run_course.id)
+        self.assertEqual(updated_dest_course.end, datetime(2020, 10, 1, tzinfo=UTC))
+        self.assertEqual(updated_dest_course.enrollment_start, datetime(2019, 10, 1, tzinfo=UTC))
+        self.assertEqual(updated_dest_course.enrollment_end, datetime(2019, 10, 10, tzinfo=UTC))
+
+    @patch('openedx.features.cms.helpers.calculate_date_by_delta')
+    def test_set_advanced_settings_due_date(self, mock_calculate_date_by_delta):
+        """
+        Testing adding valid course due date
+        """
+        re_run_course = CourseFactory.create(
+            org='org',
+            number='number',
+            run='test_dest',
+            display_name='destination course',
+            start=datetime(2019, 10, 1, tzinfo=UTC)
+        )
+
+        mock_calculate_date_by_delta.return_value = datetime(2020, 10, 1, tzinfo=UTC)
+
+        helpers.set_advanced_settings_due_date(re_run_course, mock.Mock(), self.user)
+        assert mock_calculate_date_by_delta.called
+        self.assertEqual(re_run_course.due, datetime(2020, 10, 1, tzinfo=UTC))
+
+    @patch('openedx.features.cms.helpers.calculate_date_by_delta')
+    def test_set_advanced_settings_due_date_source_course_without_due_date(
+            self, mock_calculate_date_by_delta):
+        """
+        Testing adding course due date without providing date
+        """
+        source_course = mock.Mock()
+        source_course.due = None
+        helpers.set_advanced_settings_due_date(mock.ANY, source_course, mock.ANY)
+        assert not mock_calculate_date_by_delta.called
+
+    @patch('cms.djangoapps.contentstore.tasks.apply_post_rerun_creation_tasks')
+    def test_set_rerun_module_dates(self, mock_apply_post_rerun_creation_tasks):
+        """
+        Testing start date of chapters (sequence) and start & due date of sub-sequences
+        (sequential) by creating a re-run of source course
+        """
+        source_course = test_helpers.create_source_course(self.store, self.user,
+                                                          datetime(2019, 9, 1, tzinfo=UTC))
+        result, rerun_course_id = CourseRerunFactory.create(
+            source_course_id=source_course.id,
+            user=self.user,
+            run="rerun_test",
+            start=datetime(2019, 10, 1, tzinfo=UTC)
```

Comment:
let's create a variable for this start date since we will be using it for assertion later on